client/lib/query-manager/activity/index.js

```diff
@@ -0,0 +1,78 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { overEvery, get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import QueryManager from 'lib/query-manager';
+
+/**
+ * ActivityQueryManager manages Activity which can be queried
+ */
+export default class ActivityQueryManager extends QueryManager {
+	constructor( data, options ) {
+		super( data, {
+			itemKey: 'ts_utc',
+			...options,
+		} );
+	}
+
+	/**
+	 * Sort descending order, defaulting to end of list if unknown.
+	 *
+	 * @param  {Object} query Query object (Unused).
+	 * @param  {Object} itemA First item
+	 * @param  {Object} itemB Second item
+	 * @return {Number}       0 if equal, less than 0 if itemA is first,
+	 *                        greater than 0 if itemB is first.
+	 */
+	compare( query, { ts_utc: tsA }, { ts_utc: tsB } ) {
+		return tsB - tsA;
+	}
+
+	/**
+	 * Returns true if the item matches the given query, or false otherwise.
+	 *
+	 * @param  {Object}  query Query object
+	 * @param  {Object}  item  Item to consider
+	 * @return {Boolean}       Whether item matches query
+	 */
+	matches = ActivityQueryManager.matches;
```

Comment:
> * Take a deeper look at `QueryManager` and its subclasses to see whether `matches` could/should be static. A cursory look indicates that it probably could be.

This :smile: The other ones are workarounds, this is likely the fix -- at a conceptual level, I don't think a `matches` method should ever need to refer to internal state of `QM` _by definition_, but only compare the args it's given.